contract/src/test/java/io/neow3j/contract/TransactionBuilderTest.java

```diff
@@ -174,32 +179,63 @@ public void automaticallyFetchValidUntilBlock() throws Throwable {
         assertThat(transaction.getNonce(), lessThanOrEqualTo((long) Math.pow(2, 32)));
     }
 
-    @Test(expected = IllegalStateException.class)
+    @Test
     public void failBuildingTxWithoutAnySigner() throws Throwable {
+        exceptionRule.expect(IllegalStateException.class);
+        exceptionRule.expectMessage("without any signer");
         new TransactionBuilder(neow)
                 .validUntilBlock(100L)
                 .wallet(Wallet.withAccounts(account1))
                 .script(new byte[]{1, 2, 3})
                 .buildTransaction();
     }
 
-    @Test(expected = TransactionConfigurationException.class)
-    public void failAddingMultipleSignersConcerningTheSameAccount1() {
+    @Test
+    public void failAddingMultipleSignersConcerningTheSameAccount() {
         TransactionBuilder b = new TransactionBuilder(neow);
-        b.signers(Signer.global(account1.getScriptHash()), Signer.calledByEntry(account1.getScriptHash()));
+        exceptionRule.expect(TransactionConfigurationException.class);
+        exceptionRule.expectMessage("concerning the same account");
+        b.signers(Signer.global(account1.getScriptHash()),
+                Signer.calledByEntry(account1.getScriptHash()));
     }
 
-    @Test(expected = TransactionConfigurationException.class)
-    public void failAddingMultipleSignersConcerningTheSameAccount_sequential() {
+    @Test
+    public void overrideSigner() {
         TransactionBuilder b = new TransactionBuilder(neow);
         b.signers(Signer.global(account1.getScriptHash()));
-        b.signers(Signer.calledByEntry(account1.getScriptHash()));
+        assertThat(b.getSigners(), hasSize(1));
+        assertThat(b.getSigners().get(0), is(Signer.global(account1.getScriptHash())));
+
+        b.signers(Signer.calledByEntry(account2.getScriptHash()));
+        assertThat(b.getSigners(), hasSize(1));
+        assertThat(b.getSigners().get(0), is(Signer.calledByEntry(account2.getScriptHash())));
+    }
+
+    @Test
+    public void failAddingMultipleFeeOnlySigners() {
+        TransactionBuilder b = new TransactionBuilder(neow);
+        exceptionRule.expect(TransactionConfigurationException.class);
+        exceptionRule.expectMessage(
+                new StringContains("Can't add multiple signers with the fee-only witness scope"));
+        b.signers(Signer.feeOnly(account1.getScriptHash()),
+                Signer.feeOnly(account2.getScriptHash()));
+    }
+
+    @Test
+    public void failAddingMultipleFeeOnlySignersSequentially() {
+        TransactionBuilder b = new TransactionBuilder(neow);
```

Comment:
This test should not be necessary anymore once my other comment in `TransactionBuilder` is resolved. Adding signers sequentially will simply override the last signers.